### ISSUE TITLE
Feature Request: ability to prevent loading of or cleanly unregister unused modules

### ISSUE THREAD

**Issue by mscdex** [pjsip/pjproject#3149]
**Is your feature request related to a problem? Please describe.**
Currently pjsua's sub modules do not utilize module load/unload callbacks, so it is not possible to safely unregister them.

On a related note, while pjsua exposes a public function for retrieving pjsua's `pjsip_endpoint`, there is no way to access members of that endpoint because the struct's definition is inside sip_endpoint.c instead of sip_endpoint.h. This means one needs to copy the struct definition (and any other non-public, associated structs used by `pjsip_endpoint`) in order to search for loaded modules for the purposes of unregistering them (by name).

**Describe the solution you'd like**
Either prevent the loading of non-essential modules (possibly via pjsua config) and/or be able to cleanly unregister them after pjsua has been initialized.

**Describe alternatives you've considered**
Having to maintain patches on top of pjsip? That's not something I'm really excited about...

**Additional context**
Specifically the modules I'd like to prevent from being registered or be able to cleanly unregister afterwards are:

* `mod-pjsua-im`
* `mod-pjsua-pres`
* `mod-refer`
* `mod-mwi`
* `mod-presence`
* `mod-evsub`

**Comment by bennylp** [comment]
Why do you not want these modules loaded? These modules are required for the features that PJSUA-LIB promises to provide. 

If you want to "replace" the functionality that these modules provide, you can write your own module with higher priority (=low priority number) than those modules so that your module gets called first.

**Comment by mscdex** [comment]
I wanted to avoid loading them because they were unnecessary features for my application and didn't want to waste system resources on them.

Some time after I originally submitted this issue I ended up just using the API one lower below pjsua (pjsip-ua or whatever it's called). It wasn't ideal as it meant I had to reimplement a bunch of basic stuff like bookkeeping, media handling, and other essentials, but it is what it is.
